snapcraft/internal/pluginhandler/_plugin_loader.py

```diff
@@ -74,7 +76,33 @@ def load_plugin(
     return plugin
 
 
-def _load_local(module_name: str, local_plugin_dir: str):
+def _load_compat_x_prefix(plugin_name: str, local_plugin_dir: str):
+    compat_path = Path(local_plugin_dir, f"x-{plugin_name}.py")
+    if not compat_path.exists():
+        return None
+
+    spec = importlib.util.spec_from_file_location(plugin_name, compat_path)
+    if spec.loader is None:
+        return None
+
+    # Prevent mypy type complaints by asserting type.
+    assert isinstance(spec.loader, importlib.abc.Loader)
+
+    module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(module)
+    return module
+
+
+def _load_local(plugin_name: str, local_plugin_dir: str):
+    module_name = plugin_name.replace("-", "_")
+    module = _load_compat_x_prefix(plugin_name, local_plugin_dir)
+    if module is not None:
+        suggested_name = f"x_{module_name}"
+        warnings.warn(
```

Comment:
not a fan of introducing yet another mechanism to warn (we already use logger.warning)

Reply:
Oh I just copied the approach you used for warnings.warn("Plugin import path has changed to 'snapcraft.plugins.v1') :P